snappy/security.go

```diff
@@ -121,106 +329,616 @@ func getSecurityProfile(m *packageYaml, appName, baseDir string) (string, error)
 	return fmt.Sprintf("%s.%s_%s_%s", m.Name, origin, cleanedName, m.Version), err
 }
 
-var runScFilterGen = runScFilterGenImpl
+type securityAppID struct {
+	AppID   string
+	Pkgname string
+	Appname string
+	Version string
+}
 
-func runScFilterGenImpl(argv ...string) ([]byte, error) {
-	cmd := exec.Command(argv[0], argv[1:]...)
-	return cmd.Output()
+func newAppID(appID string) (*securityAppID, error) {
+	tmp := strings.Split(appID, "_")
+	if len(tmp) != 3 {
+		return nil, errInvalidAppID
+	}
+	id := securityAppID{
+		AppID:   appID,
+		Pkgname: tmp[0],
+		Appname: tmp[1],
+		Version: tmp[2],
+	}
+	return &id, nil
 }
 
-// seccomp specific
-func generateSeccompPolicy(baseDir, appName string, sd SecurityDefinitions) ([]byte, error) {
-	if sd.SecurityPolicy != nil && sd.SecurityPolicy.Seccomp != "" {
-		fn := filepath.Join(baseDir, sd.SecurityPolicy.Seccomp)
-		content, err := ioutil.ReadFile(fn)
-		if err != nil {
-			logger.Noticef("Failed to read %q: %v", fn, err)
+// TODO: once verified, reorganize all these
+func (sa *securityAppID) appArmorVars() string {
+	aavars := "\n# Specified profile variables\n"
```

Comment:
Thanks, I changed that code to use a single fmt.Sprintf()instead of multiple string concats. If thats not sufficient I'm happy to change to use byte.Buffer (but it seems fmt.Sprintf is using a byte buffer internally already.